### ISSUE TITLE
Templates in subplots not plotting correctly in GUI

### ISSUE THREAD

**Issue by chrisoyh** [dron-tan/SpikeSimulationTool#21]
The GUI subplot extracts the largest amplitude spike it identifies from each naxon and extracts it along with a short buffer section (+/-100) around it to be used as the template

Hence, if the maximum spike happens to be part of a transitioning spike group, parts of the template displayed may contain the other transitioning spike

Also cannot identify template if there is no spike within the simulation duration.

**Comment by dron-tan** [comment]
> The GUI subplot extracts the largest amplitude spike it identifies from each naxon and extracts it along with a short buffer section (+/-100) around it to be used as the template
> 
> Hence, if the maximum spike happens to be part of a transitioning spike group, parts of the template displayed may contain the other transitioning spike
> 
> Also cannot identify template if there is no spike within the simulation duration.

Does this mean that it is only the GUI function that is affecting this? Meaning that the template that is being used is still the extracted template from gen_template.m (HHSimulation.m)?

**Comment by chrisoyh** [comment]
Yup, the template used for the simulation is still from templates_test.mat. Its just the GUI extracts the 'template' from the simulation indirectly for plotting. This was possible before because Artemio's spikes never had transitions and always had a resting period in between so the shape of the spike would be the same as the templates anyways.
